src/main/java/seedu/address/model/task/TaskDeadline.java

```diff
@@ -26,16 +30,115 @@ public TaskDeadline(String taskDeadline) {
         value = taskDeadline;
     }
 
+
+    /**
+     * Returns the time or date of the task.
+     *
+     * @param line The user's input.
+     * @return The String representation of the time or date of the task.
+     * @throws DateTimeParseException An exception thrown if the date or time is invalid.
+     */
+    public static String parseDateAndTime(Scanner line) {
+        DateTimeFormatter dateFormatter = DateTimeFormatter.ofPattern("E, MMM dd yyyy");
+        DateTimeFormatter timeFormatter = DateTimeFormatter.ofPattern("hh:mm a");
+        LocalDate date;
+        LocalTime time;
+        String formattedDateAndTime;
+        String[] split = line.nextLine().split(" ");
+        if (isInDateFormat(split[0])) {
+            date = LocalDate.parse(split[0]);
+            formattedDateAndTime = dateFormatter.format(date);
+            if (split.length == 2) {
+                time = LocalTime.parse(split[1]);
+                formattedDateAndTime = formattedDateAndTime + " " + timeFormatter.format(time);
+            }
+        } else {
+            time = LocalTime.parse(split[0]);
+            formattedDateAndTime = timeFormatter.format(time);
+            if (split.length == 2) {
+                date = LocalDate.parse(split[1]);
+                formattedDateAndTime = dateFormatter.format(date) + " " + formattedDateAndTime;
+            }
+        }
+        return formattedDateAndTime;
+    }
+
+    /**
+     * Returns whether the string is in a date format.
+     *
+     * @param string The string that may potentially be a date.
+     * @return A boolean stating whether the string is in a date format.
+     */
+    public static boolean isInDateFormat(String string) {
+        if (string.length() == 10) {
+            String year = string.substring(0, 4);
+            String month = string.substring(5, 7);
+            String day = string.substring(8, 10);
```

Comment:
For the date format, is it possible to for the parsing to stick with what we currently have (DD-MM-YYYY)? Since this affects UG/DG, some error messages and the sample data as well (all that I can think of right now). Might omit something when have so many to change :")
What do the rest think? @charliemoweng @WangGLJoseph @tanjoewel

Reply:
I actually checked the UG and DG already, and I think the deadline formatting is split pretty evenly between yyyy-mm-dd and dd-mm-yyyy hahah

Reply:
Yeah I think we should standardise it, I would suggest just standardising to DD-MM-YYYY since I think that's the most common form.

Reply:
Actually I think idk how to parse DD-MM-YYYY lol do y'all know what parser I can use to parse that?😭😭 java default parser only can parse yyyy-mm-dd:')

Reply:
Ohh in that case I think we can go with YYYY-MM-DD instead! I didn't know the default parser only parses that format

Reply:
> Actually I think idk how to parse DD-MM-YYYY lol do y'all know what parser I can use to parse that?😭😭 java default parser only can parse yyyy-mm-dd:')

Haha no worries in that case I think it looks good we can proceed with this!

Reply:
Yeah lets use YYYY-MM-DD